packages/lisk-transactions/test/transactions/base.ts

```diff
@@ -0,0 +1,746 @@
+		it('should set isMultisignature to false', () => {
+			return expect(baseTransaction.isMultisignature).to.be.false;
+		});
+	});
+
+	describe('when transaction has non-empty `signatures`', () => {
+		beforeEach(() => {
+			const multisignatureTransaction = {
+				...defaultTransaction,
+				signatures: [defaultSignature, defaultSignature],
+			};
+			baseTransaction = new TestTransaction(multisignatureTransaction);
+		});
+		it('should set isMultisignature to true', () => {
+			return expect(baseTransaction.isMultisignature).to.be.true;
+		});
+	});
+
+	describe('#constructor', () => {
+		it('should create a new instance of BaseTransaction', () => {
+			return expect(baseTransaction)
+				.to.be.an('object')
+				.and.be.instanceof(BaseTransaction);
+		});
+
+		describe('when given a transaction with invalid types', () => {
+			const invalidTransaction = {
+				...defaultTransaction,
+				amount: 0,
+				fee: 'fee',
+			};
+
+			it('should throw a transaction multierror', () => {
+				try {
+					new TestTransaction(
+						(invalidTransaction as unknown) as TransactionJSON,
+					);
+				} catch (error) {
+					return expect(error).to.be.an.instanceOf(TransactionMultiError);
+				}
+
+				return Promise.resolve();
+			});
+		});
+	});
+
+	describe('#assetToJSON', () => {
+		it('should return an object of type transaction asset', () => {
+			return expect(baseTransaction.assetToJSON()).to.be.an('object');
+		});
+	});
+
+	describe('#toJSON', () => {
+		it('should return transaction json', () => {
+			baseTransaction = new TestTransaction(defaultTransaction);
+			const transactionJSON = baseTransaction.toJSON();
+
+			return expect(transactionJSON).to.be.eql(defaultTransaction);
+		});
+	});
+
+	describe('#getBytes', () => {
+		it('should return a buffer', () => {
+			expect(baseTransaction.getBytes()).to.be.an.instanceOf(Buffer);
+		});
+	});
+
+	describe('#containsUniqueData', () => {
+		it('should return a boolean', () => {
+			expect(baseTransaction.containsUniqueData()).to.be.a('boolean');
+		});
+	});
+
+	describe('#checkSchema', () => {
+		describe('when given valid transaction', () => {
+			beforeEach(() => {
+				baseTransaction = new TestTransaction(defaultTransaction);
+
+				return Promise.resolve();
+			});
+
+			it('should return a successful transaction response', () => {
+				const { id, status, errors } = baseTransaction.checkSchema();
+
+				expect(id).to.be.eql(baseTransaction.id);
+				expect(errors).to.be.eql([]);
+				return expect(status).to.eql(Status.OK);
+			});
+		});
+
+		describe('when given invalid data', () => {
+			let invalidTransaction: any;
+			beforeEach(() => {
+				invalidTransaction = {
+					type: 0,
+					amount: '00001',
+					fee: '0000',
+					recipientId: '',
+					senderPublicKey: '11111111',
+					senderId: '11111111',
+					timestamp: 79289378,
+					asset: {},
+					signature: '1111111111',
+					id: '1',
+				};
+				baseTransaction = new TestTransaction(invalidTransaction as any);
```

Comment:
We have nested `beforeEach` functions where we are creating multiple new Transactions in each level. This will make tests a lot harder to understand and reason about. So, maybe it's better to not have so many nested levels, or only perform actions in the beforeEach blocks which are shared in all the tests.

Reply:
I can remove some new Transaction calls (the ones with `defaultTransaction`) but not all of them as it is necessary to instantiate with specific transactions in order to test methods.